pkg/virt-controller/services/template.go

```diff
@@ -1787,30 +1787,13 @@ func getVirtiofsCapabilities() []k8sv1.Capability {
 	}
 }
 
-func requireDHCP(vmi *v1.VirtualMachineInstance) bool {
-	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Bridge != nil || iface.Masquerade != nil {
-			return true
-		}
-	}
-	return false
-}
-
-func haveSlirp(vmi *v1.VirtualMachineInstance) bool {
-	for _, iface := range vmi.Spec.Domain.Devices.Interfaces {
-		if iface.Slirp != nil {
-			return true
-		}
-	}
-	return false
-}
-
 func getRequiredCapabilities(vmi *v1.VirtualMachineInstance) []k8sv1.Capability {
-	var capabilities []k8sv1.Capability
+	// These capabilies are always required because we set them on virt-launcher binary
+	// add CAP_SYS_PTRACE capability needed by libvirt + swtpm
+	// TODO: drop SYS_PTRACE after updating libvirt to a release containing:
+	// https://github.com/libvirt/libvirt/commit/a9c500d2b50c5c041a1bb6ae9724402cf1cec8fe
+	capabilities := []k8sv1.Capability{CAP_NET_BIND_SERVICE, CAP_SYS_PTRACE}
```

Comment:
[link](https://github.com/kubevirt/kubevirt/blob/main/cmd/virt-launcher/BUILD.bazel#L71) to how the caps are set in the binary.